vhdx/src/vhdx_header.rs

```diff
@@ -0,0 +1,522 @@
+// Copyright © 2021 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 AND BSD-3-Clause
+
+extern crate log;
+
+use byteorder::{ByteOrder, LittleEndian, ReadBytesExt};
+use remain::sorted;
+use std::collections::btree_map::BTreeMap;
+use std::fmt::{self, Display};
+use std::fs::File;
+use std::io::{self, Read, Seek, SeekFrom, Write};
+use std::mem::size_of;
+use uuid::Uuid;
+
+const VHDX_SIGN: u64 = 0x656C_6966_7864_6876; //"vhdxfile"
+const HEADER_SIGN: u32 = 0x6461_6568; //"head"
+const REGION_SIGN: u32 = 0x6967_6572; // "regi"
+
+const FILE_START: u64 = 0; //The first element
+const HEADER1_START: u64 = 64 * 1024; //header 1 start in Bytes
+const HEADER2_START: u64 = 128 * 1024; //header 2 start in Bytes
+const REGION1_START: u64 = 192 * 1024; // Region 1 start in Bytes
+const REGION2_START: u64 = 256 * 1024; // Region 2 start in Bytes
+
+const HEADER_SIZE: u64 = 4 * 1024; // Each header is 64 KiB, but only first 4 kiB contains info
+const REGION_SIZE: u64 = 64 * 1024; // Each region size is 64 KiB
+
+const REGION_ENTRY_REQUIRED: u32 = 1;
+
+const BAT_GUID: u128 = 11023197692918435611014362394968356710; // BAT GUID in integer read in little endian
```

Comment:
You added the uuid crate, why not use it?

Reply:
Replied in the other other comment on uuid.